### PR TITLE
Persistent RNN functions in cuDNN v6

### PR DESCRIPTION
**Pull request by aonotas**
This PR for Persistent RNN functions in cuDNN v6.

- createPersistentRNNPlan
- setPersistentRNNPlan
- destroyPersistentRNNPlan
- setRNNDescriptor_v6

### CODE REVIEW

File: cupy/cudnn.py [hunk 1]
```diff
@@ -270,16 +270,32 @@ def set_dropout_descriptor(desc, handle, dropout):
...
+            input_mode, direction, mode, algo, data_type)
+    else:
+        cudnn.setRNNDescriptor(
+            desc.value, hidden_size, num_layers, dropout_desc.value,
+            input_mode, direction, mode, data_type)
     return desc
 
 
+def create_rnn_persistent_rnn_plan(desc, minibatch, data_type):
+    # Persistent RNN
```
Comment (unnonouno) [inline cupy/cudnn.py]: Is this comment required?

**Comment by unnonouno** [comment]
jenkins test this please

**Comment by aonotas** [comment]
Thank you for your comments.
I fixed my code to pass Travis.
Please check it when you have time.

**Comment by unnonouno** [comment]
Is it possible to keep compatibility by making new arguments optional? It is because even when a user updates only CuPy, Chainer correctly works.
BTW, we need to make smarter interface.

**Comment by aonotas** [comment]
OK. 
I will fix the interface to support version mismatch between chainer and cupy.

**Comment by unnonouno** [comment]
jenkins test this please

**Comment by unnonouno** [comment]
LGTM

**Pull request merged**